eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/EventGridSubscriber.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.eventgrid.customization;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.eventgrid.models.EventGridEvent;
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.serializer.AzureJacksonAdapter;
+import com.microsoft.rest.protocol.SerializerAdapter;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The type that can be used to de-serialize events.
+ */
+@Beta
+public class EventGridSubscriber {
+    /**
+     * The default adapter for to be used for de-serializing the events
+     */
+    private final AzureJacksonAdapter defaultSerializerAdapter;
+    /**
+     * The map containing user defined mapping of eventType to Java model type
+     */
+    private Map<String, Type> eventTypeToEventDataMapping;
+
+    @Beta
+    public EventGridSubscriber() {
+        this.defaultSerializerAdapter = new AzureJacksonAdapter();
+        this.eventTypeToEventDataMapping =  new HashMap<>();
+    }
+
+    /**
+     * Add a custom event mapping. If a mapping with same eventType exists then the old eventDataType is replaced by
+     * the specified eventDataType
+     *
+     * @param eventType the event type name
+     * @param eventDataType type of the Java model that the event type name mapped to
+     */
+    @Beta
+    public void putCustomEventMapping(String eventType, Type eventDataType) {
+        if (eventType == null || eventType.isEmpty()) {
+            throw new IllegalArgumentException("eventType parameter is required and cannot be null or empty");
+        }
+        if (eventDataType == null) {
+            throw new IllegalArgumentException("eventDataType parameter is required and cannot be null");
+        }
+        this.eventTypeToEventDataMapping.put(eventType.toLowerCase(), eventDataType);
+    }
+
+    /**
+     * Get type of the Java model that is mapped to the given eventType.
+     *
+     * @param eventType the event type name
+     * @return type of the Java model id mapping exists, null otherwise
+     */
+    @Beta
+    public Type getCustomEventMapping(String eventType) {
+        if (eventType == null || eventType.isEmpty()) {
+            throw new IllegalArgumentException("eventType parameter is required and cannot be null or empty");
+        }
+        if (!this.eventTypeToEventDataMapping.containsKey(eventType.toLowerCase())) {
```

Comment:
Nit: Consider adding something like a canonicalizedEventType = eventType.toLowerCase() and then use it instead of calling toLowerCase() multiple times - in the future if new code gets added, that way it is more obvious that the lower cased string should be used.

Reply:
will do, that makes code even more readable.